src/plugins/dashboard/public/dashboard_container/embeddable/integrations/diff_state/dashboard_diffing_integration.ts

```diff
@@ -200,7 +200,11 @@ async function getInputChanges(
   await container.untilInitialized();
   const keyComparePromises = keys.map(
     (key) =>
-      new Promise<{ key: keyof DashboardContainerByValueInput; isEqual: boolean }>((resolve) =>
+      new Promise<{ key: keyof DashboardContainerByValueInput; isEqual: boolean }>((resolve) => {
+        if (input[key] === undefined && input[key] === undefined) {
```

Comment:
nit: did you mean `if (input[key] === undefined && lastInput[key] === undefined)`?

Also, just out of curiosity why include this early return, is it so we don't have to launch the async `isKeyEqual` function?

Reply:
This check was added to resolve issues in jest tests.

I like to keep jest tests as simple as possible to allow the reader to focus on cause and effect of the specific test case without all the noise of a runtime environment. This means that jest tests do not include all input fields - just input fields being tested. For example, for filter jest tests, input is just `{ filters }`.

This caused problems where `timeRange` check was causing failures with filter check test cases. The above guard was added to short circuit isKeyEqual checks if key is undefined in `input` and `lastInput`. I could have put the logic in `timeRange` diff function but I thought it made the most sense to put the guard in a more general location since it does not make sense to check anything when a key's value is undefined in both `input` and `lastInput`.